tools/tests/test_xtce_generator_template.py

```diff
@@ -0,0 +1,79 @@
+"""Testing for xtce generator template."""
+
+import sys
+from unittest import mock
+
+import pytest
+
+from tools.xtce_generation.xtce_generator_template import main
+
+
+# General test
+@mock.patch("tools.xtce_generation.xtce_generator_template.main")
+def test_main_general(mock_input):
+    """Testing base main function."""
+    test_args = [
+        "test_script",
+        "--instrument",
+        "swe",
+        "--file_path",
+        "/Users/anma6676/Desktop/Repositories/imap_processing/tools/xtce_generation/TLM_SWP_20231006-121021.xlsx",
+        "--packets",
+        '{"P_SWP_HK": 1184, "P_SWP_SCI": 1188, "P_SWP_AUT": 1192}',
+    ]
+    with mock.patch.object(sys, "argv", test_args):
+        main()
+
+
+# Testing invalid instrument
+@mock.patch("tools.xtce_generation.xtce_generator_template.main")
+def test_main_inval_instr(mock_input):
+    """Testing with invalid instrument."""
+    test_args = [
+        "test_script",
+        "--instrument",
+        "ana",
+        "--file_path",
+        "/Users/anma6676/Desktop/Repositories/imap_processing/tools/xtce_generation/TLM_SWP_20231006-121021.xlsx",
```

Comment:
I'd suggest removing your specific folder structure from tests if possible. That way the bots won't learn who you are :)
```suggestion
        "tools/xtce_generation/TLM_SWP_20231006-121021.xlsx",
```